src/prepareAssetsPackage.js

```diff
@@ -1,81 +1,51 @@
+  );
 
-    const stream = new Writable();
-    const data = [];
-
-    stream._write = (chunk, enc, done) => {
-      data.push(...chunk);
-      done();
-    };
-    stream.on('finish', () => {
-      const buffer = Buffer.from(data);
-      const hash = crypto.createHash('md5').update(buffer).digest('hex');
-      resolve({ buffer, hash });
-    });
-    archive.pipe(stream);
+  const files = Object.entries(paths).map(([assetPath, resolvePath]) => {
+    if (!resolvePath) {
+      throw new Error(`Unable to resolve asset path: ${assetPath}`);
+    }
 
-    Object.keys(paths).forEach((assetPath) => {
-      const resolvePath = paths[assetPath];
+    for (const publicFolder of publicFoldersResolved) {
+      const fullPath = path.join(publicFolder, assetPath);
+      if (fs.existsSync(fullPath)) {
+        return {
+          content: fs.createReadStream(fullPath),
+          name: resolvePath,
+        };
+      }
 
-      if (!resolvePath) {
-        throw new Error(`Unable to resolve asset path: ${assetPath}`);
+      // findCSSAssetPaths will sometimes return absolute paths
+      if (assetPath.startsWith(publicFolder) && fs.existsSync(assetPath)) {
+        return {
+          content: fs.createReadStream(assetPath),
+          name: resolvePath,
+        };
       }
 
-      for (const publicFolder of publicFolders) {
-        const folder = publicFolder.startsWith('/')
-          ? publicFolder
-          : path.resolve(process.cwd(), publicFolder);
-        const fullPath = path.join(folder, assetPath);
-        if (fs.existsSync(fullPath)) {
-          archive.append(fs.createReadStream(fullPath), {
-            name: resolvePath,
-            date: FILE_CREATION_DATE,
-          });
-          return;
-        }
-        // findCSSAssetPaths will sometimes return absolute paths
-        if (assetPath.startsWith(folder) && fs.existsSync(assetPath)) {
-          archive.append(fs.createReadStream(assetPath), {
-            name: resolvePath,
-            date: FILE_CREATION_DATE,
-          });
-          return;
-        }
-        // as a last fallback, check if the resolve path exists in the public folder
-        const fullResolvePath = path.join(folder, resolvePath);
-        if (fs.existsSync(fullResolvePath)) {
-          archive.append(fs.createReadStream(fullResolvePath), {
-            name: resolvePath,
-            date: FILE_CREATION_DATE,
-          });
-          return;
-        }
+      // as a last fallback, check if the resolve path exists in the public
+      // folder
+      const fullResolvePath = path.join(publicFolder, resolvePath);
+      if (fs.existsSync(fullResolvePath)) {
+        return {
+          content: fs.createReadStream(fullResolvePath),
+          name: resolvePath,
+        };
       }
-    });
+    }
 
-    archive.on('error', reject);
-    archive.finalize();
+    return null;
   });
+
+  return deterministicArchive([], files.filter(Boolean));
```

Comment:
Since we are generating an archive with a file structure that is different than what is on the filesystem, we need to use the second argument for everything so we can control the paths more specifically.